### PR TITLE
[FEATURE] libc++ support

### LINKED ISSUE #3207: Tests fail to compile because the C++11 feature std::result_of_t was removed in C++20
### Does this problem persist on the current master?

- [X] I have verified the issue on the current master

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```
/usr/local/bin/ccache /usr/local/llvm17/bin/clang++ -DSEQAN3_HAS_BZIP2=1 -DSEQAN3_HAS_ZLIB=1 -I/usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/test/include -I/usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/include -isystem /usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/submodules/sdsl-lite/include -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -isystem /usr/local/include -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -isystem /usr/local/include  -DNDEBUG -pedantic -Wall -Wextra -Werror -std=c++20 -pthread -MD -MT alignment/decorator/CMakeFiles/gap_decorator_test.dir/gap_decorator_test.cpp.o -MF alignment/decorator/CMakeFiles/gap_decorator_test.dir/gap_decorator_test.cpp.o.d -o alignment/decorator/CMakeFiles/gap_decorator_test.dir/gap_decorator_test.cpp.o -c /usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/test/unit/alignment/decorator/gap_decorator_test.cpp
In file included from /usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/test/unit/alignment/decorator/gap_decorator_test.cpp:19:
In file included from /usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/test/unit/alignment/decorator/../../alignment/aligned_sequence_test_template.hpp:19:
In file included from /usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/include/seqan3/io/sam_file/detail/cigar.hpp:24:
/usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/include/seqan3/io/views/detail/take_until_view.hpp:57:44: error: no member named 'result_of_t' in namespace 'std'
   57 |     static_assert(std::convertible_to<std::result_of_t<fun_t && (std::ranges::range_reference_t<urng_t>)>, bool>,
      |                                       ~~~~~^
/usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/include/seqan3/io/views/detail/take_until_view.hpp:57:56: error: 'fun_t' does not refer to a value
   57 |     static_assert(std::convertible_to<std::result_of_t<fun_t && (std::ranges::range_reference_t<urng_t>)>, bool>,
      |                                                        ^
/usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/include/seqan3/io/views/detail/take_until_view.hpp:50:46: note: declared here
   50 | template <std::ranges::view urng_t, typename fun_t, bool or_throw, bool and_consume>
      |                                              ^
/usr/ports/biology/seqan/work/seqan3-3.3.0-49-g64a431380/include/seqan3/io/views/detail/take_until_view.hpp:57:105: error: expected expression
   57 |     static_assert(std::convertible_to<std::result_of_t<fun_t && (std::ranges::range_reference_t<urng_t>)>, bool>,
      |                                                                                                         ^
3 errors generated.
```

### Expected Behavior

n/a

### Steps To Reproduce

n/a

### Environment

```markdown
- Operating system: FreeBSD 13.2
- SeqAn version: 3.3.0-49--g64a431380
- Compiler: clang-16
```

### Anything else?

This feature was removed in C++20, see [here](https://en.cppreference.com/w/cpp/types/result_of).

### PR DESCRIPTION
**Pull request by eseiler**
Resolves #3207

### CODE REVIEW

File: include/seqan3/core/debug_stream/debug_stream_type.hpp [hunk 1]
```diff
@@ -185,12 +185,17 @@ class debug_stream_type
         stream->unsetf(flag);
     }
 
+// fmtflags are an enum in libstc++ and unsigned in libc++
```
Comment (eseiler) [inline include/seqan3/core/debug_stream/debug_stream_type.hpp:188]:
```suggestion
// fmtflags are an enum in libstdc++ and unsigned in libc++
```

File: include/seqan3/io/sequence_file/format_embl.hpp [hunk 2]
```diff
@@ -105,12 +105,14 @@ class format_embl
                               id_type & id,
                               qual_type & SEQAN3_DOXYGEN_ONLY(qualities))
     {
+        // Store current position in buffer
+        // Must happen before construction the view.
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_embl.hpp:109]:
```suggestion
        // Must happen before constructing the view.
```
And others

File: test/unit/test/expect_same_type_test.cpp [hunk 3]
```diff
@@ -161,12 +161,19 @@ TEST(tuple, same_type_pass)
 
 TEST(tuple, same_type_fail)
 {
+#ifdef _LIBCPP_VERSION
+    char const * error_message = "Expected equality of these values:\n"
+                                 "  decltype(std::tuple{0, .0f, .0, 0u})\n"
+                                 "    Which is: \"std::__1::tuple<int, float, double, unsigned int>\"\n"
+                                 "  std::tuple<int, float, unsigned, double>\n"
```
Comment (eseiler) [inline test/unit/test/expect_same_type_test.cpp:168]: Rename commit

**Review by SGSSGene — APPROVED** [review]
Great!

**Pull request merged**